File: client/src/shared/models/BugCreationRequestPayload.ts
import { StatusPayload } from './StatusPayload';
import { BugPayload } from './BugPayload';

export default class BugCreationRequestPayload {
    name: string;
    description: string;
    status: StatusPayload | null;

    constructor() {
        this.name = '';
        this.description = '';
        this.status = null;
    }

    static from(bug: BugPayload): BugCreationRequestPayload {
        const request: BugCreationRequestPayload = new BugCreationRequestPayload();
        request.name = bug.name;
        request.description = bug.description;
        request.status = bug.status;
        return request;
    }
}

File: api/src/main/java/com/jlopez/bugtracker/model/BugCreationRequestPayload.java
package com.jlopez.bugtracker.model;

import java.io.Serializable;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class BugCreationRequestPayload implements Serializable {

  private static final long serialVersionUID = 5873297484454588086L;

  @Size(max = 255, message = "The title cannot be longer than 255 characters")
  private final String name;

  private final String description;

  @NotNull
  private final StatusPayload status;
}
